syntax = "proto3";

package fbsrankings.messages.query;

import "fbsrankings/messages/enums/enums.proto";
import "fbsrankings/messages/options/options.proto";

message AffiliationBySeasonResult {
    string affiliation_id = 1;
    string season_id = 2;
    int32 year = 3;
    string team_id = 4;
    string team_name = 5;
    fbsrankings.messages.enums.Subdivision subdivision = 6;
}

message AffiliationsBySeasonResult {
    string query_id = 1;
    repeated AffiliationBySeasonResult affiliations = 2;
}

message AffiliationsBySeasonQuery {
    string query_id = 1;
    string season_id = 2;

    option (fbsrankings.messages.options.topic) = "fbsrankings.query.affiliations_by_season";
}
